src/hugo/content/en-us/implementations/aws/device_bootstrap/aws-iot-certificate-vending-machine/_index.md

```diff
@@ -0,0 +1,425 @@
+
+```json
+{
+    "device_uid": "DEVICE001",
+    "device_token": "<DEVICE_TOKEN>",
+    "is_enabled": "1",
+    "is_registered": "0"
+}
+```
+
+Replace `API_ENDPOINT` with your API Gateway endpoint and perform a device registration API call as acting like your device's firmware.
+
+```bash
+curl -d '{"device_uid":"DEVICE001", "device_token":"d2008..."}' -H "Content-Type: application/json" -X POST https://API_ENDPOINT/registration
+
+{"statusCode": 200, "status": "SUCCESS", "payload": {...}}
+```
+
+You registered the device. The response contains the connection info and certificates. This response is only available once so make sure that you saved the HTTP response. Later requests will responded as _device is already registered._
+
+Check the DynamoDB record to see the device's updated info.
+
+Device is registered to AWS IoT Core using certificate vending machine method. You can navigate to [AWS IoT Core Console](https://console.aws.amazon.com/iot/home) to see the newly created IoT thing and its attached resources.
+
+## Considerations
+
+This implementation covers the basics of an IoT certificate vending machine method. It does not cover certain aspects that may arise in production use.
+
+### Security and Authentication
+If your IoT devices are exposed to the internet instead of an intranet, your API should be deployed as a public API endpoint. This will allow any internet-connected network to connect your API but also it exposes your API endpoint to possible attacks. To protect your API Gateway endpoint, use [Security in Amazon API Gateway
+](https://docs.aws.amazon.com/apigateway/latest/developerguide/security.html) documentation to enable features like API rate limiting or IP address-based restrictions with AWS WAF.
+
+This implementation depends on a single unique identifier of a device. If your device unique identifiers are sequential or easy-to-estimate, you need to consider the leakage of your device token calculation formula and hash string. For this reason, you may need to implement a second unique identifier to mitigate the replay or increment attacks. In this situation, the device should claim its first unique ID, matching second unique ID, and matching calculated hash, for example `device_token=SHA256("YOUR_SECRET_SALT//DEVICE_UID_1+DEVICE_UID_2")` to be registered.
+
+Your `device_token` is calculated by hashing functions using your secret salt string. You need to implement some extra layers to increase the protection of registration logic and protect your sensitive material. 
+- If your device's computing environment provides hardware-based encryption blocks or read-protected memory blocks, you can leverage these features to reduce the possible attack surface. Almost all modern MCU's provide these feature sets out of the box, otherwise, you may need to include a TPM (Trusted Platform Module) component to your hardware design.
```

Comment:
TPM is needed if the first sentence isn't the case. The MCU statements is removed:

>If your device's computing environment provides hardware-based encryption blocks or read-protected memory blocks, you can leverage these features to reduce the possible attack surface. Otherwise, you may need to include a TPM (Trusted Platform Module) component to your hardware design.